3-extra/1-choose-your-own-adventure.js

```diff
@@ -53,6 +55,9 @@ let game = {
     //
     // Hint: the room objects have north/east/south/west methods which return
     // a new room object that is in the relevant direction.
+    this.currentRoom =this.currentRoom[direction]
+    return this.currentRoom.name
+
```

Comment:
this is correct. You only miss a step.  
let's take example of:
```
hall: {
    name: "hall",
    north: function () {
      return null;
    },
    east: function () {
      return rooms.classroom;
    },
    south: function () {
      return rooms.library;
    },
    west: function () {
      return null;
    }
```

when `this.currentRoom[direction]`  so I.E. `room['west']` is `null` it means you have an invalid movement.
So you could write a statement where if `this.currentRoom[direction]` is `null` so falsey you log an error, otherwise you use the code you have

Reply:
still did not get it :)
will see proper solution next week